extensions-core/protobuf-extensions/src/main/java/org/apache/druid/data/input/protobuf/ProtobufInputRowParser.java

```diff
@@ -127,49 +127,59 @@ void initDescriptor()
 
   private Descriptor getDescriptor(String descriptorFilePath)
   {
-    InputStream fin;
-
-    fin = this.getClass().getClassLoader().getResourceAsStream(descriptorFilePath);
+    InputStream fin = this.getClass().getClassLoader().getResourceAsStream(descriptorFilePath);
     if (fin == null) {
-      URL url;
       try {
-        url = new URL(descriptorFilePath);
+        final URL url = new URL(descriptorFilePath);
+        fin = url.openConnection().getInputStream();
       }
       catch (MalformedURLException e) {
         throw new ParseException(e, "Descriptor not found in class path or malformed URL:" + descriptorFilePath);
       }
-      try {
-        fin = url.openConnection().getInputStream();
-      }
       catch (IOException e) {
-        throw new ParseException(e, "Cannot read descriptor file: " + url);
+        throw new ParseException(e, "Cannot read descriptor file: " + descriptorFilePath);
       }
     }
 
-    DynamicSchema dynamicSchema;
+    final DescriptorProtos.FileDescriptorProto schema;
     try {
-      dynamicSchema = DynamicSchema.parseFrom(fin);
-    }
-    catch (Descriptors.DescriptorValidationException e) {
-      throw new ParseException(e, "Invalid descriptor file: " + descriptorFilePath);
+      final DescriptorProtos.FileDescriptorSet set = DescriptorProtos.FileDescriptorSet.parseFrom(fin);
+      schema = set.getFile(0);
     }
     catch (IOException e) {
       throw new ParseException(e, "Cannot read descriptor file: " + descriptorFilePath);
     }
+    finally {
+      CloseQuietly.close(fin);
+    }
 
-    Set<String> messageTypes = dynamicSchema.getMessageTypes();
-    if (messageTypes.size() == 0) {
+    final Descriptors.FileDescriptor fd;
+    try {
+      fd = Descriptors.FileDescriptor.buildFrom(schema, new Descriptors.FileDescriptor[] {});
```

Comment:
It would be nice to make `new Descriptors.FileDescriptor[] {}` as a static final value.